plugin/src/init.server.lua

```diff
@@ -145,73 +169,24 @@ function connectServer()
-frame.BackgroundColor3 = theme:GetColor(Enum.StudioStyleGuideColor.MainBackground)
-frame.Size = UDim2.fromScale(1, 1)
-frame.Parent = widget
-
-local listLayout = Instance.new("UIListLayout")
-listLayout.Padding = UDim.new(0, 5)
-listLayout.SortOrder = Enum.SortOrder.LayoutOrder
-listLayout.Parent = frame
-
-local padding = Instance.new("UIPadding")
-padding.PaddingTop = UDim.new(0, 5)
-padding.PaddingBottom = UDim.new(0, 5)
-padding.PaddingLeft = UDim.new(0, 5)
-padding.PaddingRight = UDim.new(0, 5)
-padding.Parent = frame
-
-local portTextBox = Instance.new("TextBox")
-portTextBox.BackgroundColor3 = theme:GetColor(Enum.StudioStyleGuideColor.InputFieldBackground)
-portTextBox.BorderColor3 = theme:GetColor(Enum.StudioStyleGuideColor.InputFieldBorder)
-portTextBox.TextColor3 = theme:GetColor(Enum.StudioStyleGuideColor.MainText)
-portTextBox.PlaceholderColor3 = theme:GetColor(Enum.StudioStyleGuideColor.SubText)
-portTextBox.PlaceholderText = "Port"
-portTextBox.Text = port
-portTextBox.TextSize = 20
-portTextBox.Size = UDim2.new(1, 0, 0, 30)
-portTextBox.LayoutOrder = 0
-portTextBox.Parent = frame
-portTextBox.ClearTextOnFocus = false
-portTextBox:GetPropertyChangedSignal("Text"):Connect(function()
-	portTextBox.Text = portTextBox.Text:gsub("%D+", ""):sub(1, 5)
-	port = tonumber(portTextBox.Text)
-	plugin:SetSetting("Port", port)
-end)
+ConnectButton.Click:Connect(connectServer)
 
-local connectButton = Instance.new("TextButton")
-connectButton.BackgroundColor3 = theme:GetColor(Enum.StudioStyleGuideColor.MainButton)
-connectButton.BorderColor3 = theme:GetColor(Enum.StudioStyleGuideColor.ButtonBorder)
-connectButton.TextColor3 = theme:GetColor(Enum.StudioStyleGuideColor.ButtonText)
-connectButton.Text = if connected.Value then "Disconnect" else "Connect"
-connectButton.TextSize = 16
-connectButton.Size = UDim2.new(1, 0, 0, 25)
-connectButton.LayoutOrder = 1
-connectButton.Parent = frame
-connectButton.Activated:Connect(connectServer)
+ConnectAction.Triggered:Connect(connectServer)
 
 connected.Changed:Connect(function()
-	connectButton.Text = if connected.Value then "Disconnect" else "Connect"
-	button.Icon = if connected.Value then "rbxassetid://11116536087" else "rbxassetid://11115506617"
+	ConnectButton.Icon = if connected.Value then "rbxassetid://11116536087" else "rbxassetid://11115506617"
 end)
 
-local corner = Instance.new("UICorner")
-corner.CornerRadius = UDim.new(0, 2)
-corner.Parent = portTextBox
-
-local corner2 = Instance.new("UICorner")
-corner2.CornerRadius = UDim.new(0, 2)
-corner2.Parent = connectButton
-
-button.Click:Connect(function()
-	widget.Enabled = not widget.Enabled
-end)
+if Settings and Settings.startAutomatically then
+	connectServer()
+end
 
-widget:GetPropertyChangedSignal("Enabled"):Connect(function()
-	button:SetActive(widget.Enabled)
+SettingsModule.Changed:Connect(function()
+	if connected.Value then
+		print("[Luau Language Server] Disconnecting from DataModel changes")
+		cleanup()
+	end
+	LoadSettings()
```

Comment:
Should we automatically reconnect if we were originally connected?

Reply:
We can check if user was connected when opened the settings module and auto reconnect, only downside is it spams to output a bit since event triggers on every character change which is why I didn't do at the first place

Reply:
Oh I didn't realise it was every character change. Thought it was on save, but now I remember there isn't really such thing in studio

Reply:
Looks like it doesn't cause any issue other than output spamming so I think we can solve with adding a silent mode to connect function (just for auto reconnect) which is fine since we would still have warnings for invalid setting types